src/engine/MaxConstraint.cpp

```diff
@@ -406,8 +374,14 @@ bool MaxConstraint::satisfied() const
     if ( !( _assignment.exists( _f ) && _assignment.size() > 0 ) )
         throw MarabouError( MarabouError::PARTICIPATING_VARIABLES_ABSENT );
 
+    auto byAssignment = [&](const unsigned& a, const unsigned& b) {
+                            return _assignment[a] < _assignment[b];
+                        };
+    double maxValue =  _assignment.get( *std::max_element( _elements.begin(),
```

Comment:
I think this line is getting the *element* with the maximal assignment?

Reply:
Still not convinced. std::max_element doesn't distinguish between max's f and its elements, no?
I think if you have f=max(x1,x2), and currently f=7, x1=x2=1, you would get maxValue = 7.

Reply:
I still think the code is correct, since "_elements" contains only input variables to Max and does not include _f. But I do think the code is more confusing than it needs to be. I got rid of the std::max_element business.